app/ide-desktop/lib/client/src/bin/server.ts

```diff
@@ -125,13 +133,27 @@ export class Server {
                             `Serving files from '${path.join(process.cwd(), this.config.dir)}'.`
                         )
                         resolve()
+                        if (process.env.ELECTRON_DEV_MODE === 'true') {
+                            // eslint-disable-next-line no-restricted-syntax, @typescript-eslint/consistent-type-imports, @typescript-eslint/no-var-requires
+                            const vite = require('vite') as typeof import('vite')
+                            void vite
+                                .createServer({
+                                    server: {
+                                        middlewareMode: true,
+                                        hmr: httpServer ? { server: httpServer } : {},
+                                    },
+                                    configFile: process.env.GUI2_CONFIG_PATH ?? false,
+                                })
+                                .then(devServer => (this.devServer = devServer))
+                        }
                     })()
                 }
             )
         })
     }
 
-    /** Respond to an incoming request. */
+    /** Respond to an incoming request.
+     * @throws {Error} on malformed input. */
```

Comment:
Might be helpful to explain what `malformed input` means in this case, or provide an example